app/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnector.scala

```diff
@@ -89,10 +92,16 @@ class EISConnectorImpl(
       (t: Either[RoutingError, Unit]) => Future.successful(t.isRight),
       onFailure
     ) {
+      val data          = messageSender.value.split("-")
+      var correlationId = UUID.randomUUID().toString
```

Comment:
What's the decision around using a random uuid vs setting it as expected with movement and message id?

Reply:
Also, a more functional way to write this would be:

val correlationId = messageSender.value.split("-") match {
case 2 => CorrelationId(MovementId(data(0)), MessageId(data(1))).value
case _ => UUID.randomUUID().toString
}

using var is disencouraged

Reply:
updated the code - for the question, used random uuid only if the Message Sender doesn't have the data for some reason.

Reply:
The message sender object is constructed from the movement and message id, so it will always have the data.Also we wouldn't want to be sending a random value for that as we will be using it to match up movements.

object MessageSender {

  def apply(movementId: MovementId, messageId: MessageId): MessageSender =
    MessageSender(s"${movementId.value}-${messageId.value}")
}

Can you please just update it to:

val correlationId = CorrelationId(messageSender.movementId, messageSender.messageId)